libs/utils/trace.py

```diff
@@ -629,6 +631,44 @@ def _sanitize_SchedOverutilized(self):
         self._log.debug('Overutilized time: %.6f [s] (%.3f%% of trace time)',
                         self.overutilized_time, self.overutilized_prc)
 
+    def _sanitize_ThermalTemperature(self):
+        if not self.hasEvents('thermal_temperature'):
+            return
+
+        df = self._dfg_trace_event('thermal_temperature')
+
+        # Temp scale is in milliCelcius
+        temp = df['temp'] / 1000
+        df.loc[df.index, 'temp'] = temp
```

Comment:
- Isn't this `df['temp'] /= 1000`?
- What's wrong with milliCelcius anyway? Changing units just seems like a way to confuse people.

Reply:
> Isn't this df['temp'] /= 1000?
Yeah totally. I kept getting a SettingWithCopy warning, and even though that warning was a false positive, I tried to make sure it didn't get fired - that `loc` thing was the cleanest solution I found.

Now, I've re-run the code that directly divides temperature and I don't get that warning anymore...

> What's wrong with milliCelcius anyway?
That's how the data was handled in TRAPpy, I didn't look any further and reproduced it here.

Reply:
I like better the human readable Celsius... this I'm all for the TRAPpy proposed conversion.

Reply:
Argh, I removed it in my last push :(

Reply:
Hmm, I really don't like the idea of changing units for readability when it affects internal code too - if millicelcius is hard to read then that should be solved when presenting the data (i.e. when you plot it) IMO. But if TRAPpy was already doing it, then let's keep it the same.

Reply:
> But if TRAPpy was already doing it, then let's keep it the same.

Poor choice of words on my part, sorry - the internal dataframe wasn't modified, the milliCelcius -> Celcius conversion was only done for plotting. To be honest, I don't think milliCelcius is that bad - besides, everything temperature-related is in this unit scale (IPA trip points, sensor feedback...)

Reply:
> Celcius conversion was only done for plotting

Oh, if that's true then I didn't read the code properly and I'm happy with the conversion.

Reply:
> Oh, if that's true then I didn't read the code properly and I'm happy with the conversion.

This is a wonderful case of "quiproquo" - I was describing how it was done in TRAPpy, not how I was doing it.
- TRAPpy: only convert to Celcius for plotting
- First version of this PR: Convert internal df to Celcius (affects plotting & analysis)
- Suggested course of action: Don't even convert even for plotting (but maybe add a 'milliCelcius' ylabel).